st2api/st2api/controllers/v1/keyvalue.py

```diff
@@ -116,21 +117,22 @@ def get_all(self, prefix=None, scope=FULL_SYSTEM_SCOPE, user=None, decrypt=False
             scope = FULL_USER_SCOPE
 
         scope = get_datastore_full_scope(scope)
-        requester_user = get_requester()
+        requester_user = cfg.CONF.system_user.user
```

Comment:
I have a feeling you've simplified this part too much. In this controller, we're dealing with two kinds of user: one requests the key and another one is on behalf of whom the key is requested. You provide username for the second one through `user` query parameter. First one, according to `get_requester()` is either user from auth context (one you've authenticated with) or system_user (stanley) for the `auth=false` case.

To fix that, you would need to get second user from WSGI context like https://github.com/StackStorm/st2/blob/feature/rest_services_v2/st2api/st2api/controllers/openapi.yaml#L408-L411. Problem is, now they have conflicting variable names.